<launch>
  <node name="nmea_gps_driver" pkg="nmea_gps_driver" type="nmea_gps_driver.py">
    <param name="port" value="/dev/ttyACM1"/>
    <param name="frame_id" value="base_gps" />
    <remap from="/fix" to="/gps_fix" />
    <remap from="/vel" to="/gps_velocity" />
  </node>

	<node name="p2os" pkg="p2os_driver" type="p2os" >
		<param name="pulse" value="1.0" />
		<!--<remap from="pose" to="odom" />-->
  </node>
  
	<node name="p2os_dashboard" pkg="p2os_dashboard" type="p2os_dashboard" >
  </node>

  <node name="tf_broadcaster" pkg="learning_tf.callum" type="tf_broadcaster" >
  	<param name="tilt_pitch" value="0.5828" />
  	<param name="tilt_height" value="0.28" />
  </node>

  <node name="hector_mapping" pkg="hector_mapping" type="hector_mapping" > 
  	<param name="map_frame" value="map" />
  	<remap from="map" to="map"/>
  	<remap from="syscommand" to="syscommand"/>
  	<remap from="slam_out_pose" to="nav_slam_out_pose"/>
  	<remap from="poseupdate" to="poseupdate"/>
  	<param name="base_frame" value="base_link" />
  	<param name="odom_frame" value="odom" />
  	<param name="pub_map_odom_transform" value="true" />
  	<param name="pub_map_scanmatch_transform" value="false" />
  	<param name="tf_scan_match_transform_frame_name" value="base_link" />
  	<param name="map_size" value="1500" />
  	<param name="map_pub_period" value="5.0" />
  	<param name="map_resolution" value="0.05" />
  	<param name="map_update_distance_thresh" value="0.1" />
  	<param name="map_update_angle_thresh" value="0.1" />
  </node>

  <node name="LMS100" pkg="LMS1xx" type="LMS100" > 
    <param name="host" value="192.168.1.3"/> <!-- Make sure to change this for different robots -->
		<param name="frame_id" value="base_laser" />
  </node>

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="mjpeg" />
    <param name="camera_frame_id" value="base_camera" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name="hokuyo_node" pkg="hokuyo_node" type="hokuyo_node"> 
  	<remap from="scan" to="tilt_scan"/>
  	<param name="frame_id" value="tilt_laser" />
  </node>

	<node pkg="move_base" type="move_base" name="move_base" output="screen">
		<param name="footprint_padding" value="0.01" />
    <param name="controller_frequency" value="10.0" />
    <param name="controller_patience" value="3.0" />

    <param name="oscillation_timeout" value="30.0" />
    <param name="oscillation_distance" value="0.5" />
    <param name="oscillation_timeout" value="10.0" />
    <param name="conservative_reset_dist " value="0.01" />

    <param name="TrajectoryPlannerROS/max_vel_x" value="0.6" />
    
		<rosparam file="/home/wambot/fuerte_workspace/run_wambotX/costmap_common_params.yaml" command="load" ns="global_costmap" />
		<rosparam file="/home/wambot/fuerte_workspace/run_wambotX/costmap_common_params.yaml" command="load" ns="local_costmap" />
		<rosparam file="/home/wambot/fuerte_workspace/run_wambotX/local_costmap_params.yaml" command="load" />
		<rosparam file="/home/wambot/fuerte_workspace/run_wambotX/global_costmap_params.yaml" command="load" />
		<rosparam file="/home/wambot/fuerte_workspace/run_wambotX/base_local_planner_params.yaml" command="load" />

    <remap from="/odom" to="/goal_provider_node/odom" />
  </node>

  <node pkg="hector_pose_estimation" type="pose_estimation" name="pose_estimation">
	  <param name="world_frame" value="map" />
	  <param name="nav_frame" value="odom" />
	  <param name="base_frame" value="base_link" />
	  <param name="footprint_frame" value="" />
	  <param name="position_frame" value="" />
	  <param name="stabilized_frame" value="" />
    <remap from="/fix" to="/gps_fix" />
    <remap from="/fix_velocity" to="/gps_velocity" />
    <remap from="/velocity" to="/hl_velocity" />
    <remap from="/local_poseupdate" to="/poseupdate" />
  </node>

  <node name="goal_provider_node" pkg="goal_provider" type="goal_provider_node" output="screen">
    <param name="callbacks_till_goal_update" value="1000"/>
    <param name="goal_attempt_limit" value="10" />
    <param name="subgoal_distance" value="10" />
    <param name="testing" value="true" />
    <param name="using_imu" value="true" />
    <param name="path" value="/home/wambot/fuerte_workspace/" />
  </node>
</launch>
